addons/hw_drivers/iot_handlers/drivers/L10nKeEDISerialDriver.py

```diff
@@ -188,16 +204,16 @@ def abort_post(self):
         open otherwise, blocking further invoices being sent.
         """
         self.message_number += 1
-        abort = struct.pack('BBB', 37, self.message_number + 32, 0x39)
+        abort = struct.pack('BBB', 35, self.message_number + 32, 0x39)
         request = struct.pack('B3s2sB', STX, abort, self.generate_checksum(abort), ETX)
-        time.sleep(self._protocol.commandDelay)
         self._connection.write(request)
-        time.sleep(self._protocol.measureDelay)
-        response = self._connection.read_all()
+        response = self._connection.read(COMMAND_OUTPUT_SIZE[0x39])
```

Comment:
`response = self._connection.read(COMMAND_OUTPUT_SIZE[0x39])`
to
`response = self._connection.read_all()`

Reply:
This is a case where we seriously benefit from using `read`. 

If we abort an open invoice, then the device takes a variable amount of time to cancel depending on the size of the open invoice.

With one line on the invoice, the device my cancel, and write to the buffer in 0.1 seconds

With thirty lines, it can take over 3 seconds. 

We don't always want to wait 3 seconds before reading the buffer. If we know the expected size of the output (which is `COMMAND_OUTPUT_SIZE[0x39]`), then we can use `read` rather than `read_all`.

This benefits us in 2 ways, 
Firstly, if an invoice is much faster to close than 3 seconds (maybe the invoice is just one line), then we don't wait the entire 3 seconds. 

Secondly, if it's a big invoice, then waiting 3 seconds might not be enough for the device to write to the buffer. So when we call `read_all` we get an empty byte string `b''`. If we use `read` however, the function will return when the buffer has been filled (to the `EXPECTED_OUTPUT_SIZE[0x39]`) or it will eventually time out.


When using `read_all` we have to hope that it's been long enough for the device to process, or we might wait much longer than is required.
When using `read`.. when we know the output size (which is most cases), we wait precisely however long the device needs. It's a really significant performance boost.